lib/active_remote/cached/cache.rb

```diff
@@ -42,11 +58,23 @@ def fetch(name, options = {})
 
     def read(*args)
       nested_cache_provider.read(*args) || super
+    rescue => e
+      raise e unless ::ActiveRemote::Cached.default_options[:handle_cache_error]
+      error_proc = ::ActiveRemote::Cached.default_options[:cache_error_proc]
+      error_proc.call(e) if error_proc.respond_to?(:call)
+
+      nil
     end
 
     def write(*args)
       nested_cache_provider.write(*args)
       super
+    rescue => e
+      raise e unless ::ActiveRemote::Cached.default_options[:handle_cache_error]
```

Comment:
Might be worth deduping this to say:

```
rescue => e
  handle_or_reraise_cache_error(e)
```

Since this is copied 4 times in this one file. Doesn't really impact this diff, but it's easy to miss an update down the line.